internal/tidy/tidy_test.go

```diff
@@ -0,0 +1,122 @@
+package tidy_test
+
+import (
+	"go.arcalot.io/assert"
+	"go.flow.arcalot.io/engine/internal/tidy"
+	"sort"
+	"testing"
+)
+
+func TestUnnestLongerSortedHappy(t *testing.T) {
+	astromech := []string{"q7", "bb", "r2", "r4"}
+	protocol := []string{"c3po", "000", "talky"}
+	battle := []string{"b1", "ig"}
+	probe := []string{"viper"}
+	astromechGroup := "astromech"
+	protocolGroup := "protocol"
+	battleGroup := "battle"
+	probeGroup := "probe"
+	astromechSorted := make([]string, len(astromech))
+	protocolSorted := make([]string, len(protocol))
+	battleSorted := make([]string, len(battle))
+	probeSorted := make([]string, len(probe))
+	sort.Strings(astromechSorted)
+	sort.Strings(battleSorted)
+	sort.Strings(probeSorted)
+	sort.Strings(protocolSorted)
+	expOut := [][]string{
+		{"astromech", "bb"},
+		{"astromech", "q7"},
+		{"astromech", "r2"},
+		{"astromech", "r4"},
+		{"battle", "b1"},
+		{"battle", "ig"},
+		{"probe", "viper"},
+		{"protocol", "000"},
+		{"protocol", "c3po"},
+		{"protocol", "talky"},
+	}
+	input := map[string][]string{
+		protocolGroup:  protocol,
+		astromechGroup: astromech,
+		battleGroup:    battle,
+		probeGroup:     probe,
+	}
+	assert.Equals(t, tidy.UnnestLongerSorted(input), expOut)
+}
+
+func TestUnnestLongerSortedEmpty(t *testing.T) {
+	input := map[string][]string{}
+	assert.Equals(t, tidy.UnnestLongerSorted(input), [][]string{})
+}
+
+func TestUnnestLongerSortedEmptyGroup(t *testing.T) {
+	input := map[string][]string{
+		"G": {"g"},
+		"D": {},
+	}
+	expOut := [][]string{
+		{"G", "g"},
+	}
+	assert.Equals(t, tidy.UnnestLongerSorted(input), expOut)
+}
+
+func TestSwapColumnsHappy(t *testing.T) {
+	testData := map[string]struct {
+		input    [][]string
+		expected [][]string
+	}{
+		"happy": {
+			input: [][]string{{"fruit", "tomato"},
+				{"fruit", "cucumber"},
+				{"veggie", "spinach"},
+				{"veggie", "carrot"},
```

Comment:
There's no such thing as a "vegetable" ... spinach is a leaf and carrot is a root. Yeah, OK, too much ...